src/engine/ui/widgets/Slider.hxx

```diff
@@ -67,11 +75,14 @@ private:
   /// minimum slider value
   int m_minVal = 0;
   /// maximum slider value
-  int m_maxVal = 100;
+  int m_maxVal = 0;
   /// current slider value
   int curVal;
   /// whether or not the button is to follow the mouse
   bool dragMode;
+
+  // the elements to scroll
+  std::vector<UIElement *> m_uiElements;
```

Comment:
again, the widget is for drawing only. it must not store any information about a specific usecase.

Reply:
This must be done. When a frame widget is implemented it will have to do this. It will not be able to operate without knowing what elements it contains.